### PR TITLE
Reader Recent Overhaul: Add onboarding

### PR DESCRIPTION
**Pull request by DustyReagan**
Related to https://github.com/Automattic/loop/issues/201

## Proposed Changes

* This PR displays the Reader Onboarding component and an intro message if the user has no recent feed data.
* It introduces a `forceShow` prop on the `ReaderOnboarding` component so that it ignores the component's display criteria and forces it to show in this empty recent feeds context.
* It refreshes the recent feed data when the user clicks the final "Continue" button in Reader Onboarding. This closes Reader Onboarding and shows the user their new recent feed.
* It also adds a check to the Recent component that checks if the recent feed data has loaded for the first time before rendering any content. This prevents the flashing of components that are not in context while data is loading.

Demo:

https://github.com/user-attachments/assets/b0f255bb-c4a9-48d0-88bd-14388a5addac

## Why are these changes being made?

* Adds onboarding for users who are not yet following any blogs.

## Testing Instructions

* With an account that is not following any blogs
* Use Calypso Live and go to /read?flags=reader/recent-feed-overhaul
* Interact with reader onboarding and try to break it
* Once you're following a blog, reader onboarding should not be visible on /read?flags=reader/recent-feed-overhaul

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/reader/recent/index.tsx [hunk 1]
```diff
@@ -147,50 +149,88 @@ const Recent = () => {
 		}
 	}, [ isWide, data?.items, selectedItem ] );
 
+	const [ hasLoadedItems, setHasLoadedItems ] = useState( false );
+
+	useEffect( () => {
+		const isEmpty = ! data?.items || data.items.length === 0;
```
Comment (holdercp) [inline client/reader/recent/index.tsx]: Could we move this outside and "above" of the `useEffect` callback and then remove the duplicate statement on line 166?

**Review by holdercp — APPROVED** [review]
Works as expected. One question/suggestion.

File: client/reader/recent/index.tsx (hunk 1 above)
Comment (DustyReagan) [reply client/reader/recent/index.tsx]: That makes sense to me! I updated the code in 33cce9e04576b32b8a9493b44834643855ac3709

**Comment by holdercp** [comment]
@DustyReagan One thing I'm noticing as I'm working on Automattic/loop#193 and handling "No Posts" in the feed is that the onboarding flow shows up when you're subscribed to feeds but they have no posts.
image
We should probably only show this flow when the user isn't subscribed vs. subscribed but a feed has no posts.

**Comment by DustyReagan** [comment]
> @DustyReagan One thing I'm noticing as I'm working on [Automattic/loop#193](https://github.com/Automattic/loop/issues/193) and handling "No Posts" in the feed is that the onboarding flow shows up when you're subscribed to feeds but they have no posts. ... We should probably only show this flow when the user isn't subscribed vs. subscribed but a feed has no posts.

I see what you mean. Can I get your thoughts on this newest commit, @holdercp? 47604d97c9448230d25d5f0e5ae8638c7db9306e

**Review by holdercp — APPROVED** [review]
New changes look good 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17002220

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @DustyReagan, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `{{strong}}Welcome!{{/strong}} Follow your favorite sites and their latest posts will appear here. Read, like, and comment in a distraction-free environment. Get started by selecting your interests below:`

Thank you in advance!

**Pull request merged**